File: 1392-find-the-difference-of-two-arrays/1392-find-the-difference-of-two-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[][]}
 */
var findDifference = function(nums1, nums2) {
    const set1 = new Set(nums1.map(Number));
    const set2 = new Set(nums2.map(Number));
    console.log(set1)
    const distinctsum1 = [];
    const distinctsum2 = [];
    for (const num of set1){
        if(!set2.has(num)){
            distinctsum1.push(num)
        }

    }
    for (const num of set2){
        if(!set1.has(num)){
            distinctsum2.push(num)
        }
    }

    return [distinctsum1, distinctsum2]
    
};

File: 1392-find-the-difference-of-two-arrays/1392-find-the-difference-of-two-arrays.py
class Solution(object):
    def findDifference(self, nums1, nums2):
       nums1 = set(nums1)
       nums2 = set(nums2)
       distinct_set1 = list(nums1-nums2)
       distinct_set2 = list(nums2- nums1)
       return [distinct_set1, distinct_set2]